contracts/BinaryOption.sol

```diff
@@ -19,125 +18,50 @@ contract BinaryOption is IERC20, IBinaryOption {
-        // should never occur given the minimum bid size.
-        if (exercisableDeposits <= _totalSupply) {
-            return 0;
-        }
-        return exercisableDeposits.sub(_totalSupply);
-    }
+    bool public initialized = false;
 
-    function totalClaimableSupply() external view returns (uint) {
-        (, uint exercisableDeposits) = market.senderPriceAndExercisableDeposits();
-        return _totalClaimableSupply(exercisableDeposits);
+    function initialize(
+        string memory _name,
+        string memory _symbol
+    ) public {
+        require(!initialized, "Binary Option Market already initialized");
+        initialized = true;
+        name = _name;
+        symbol = _symbol;
+        market = BinaryOptionMarket(msg.sender);
     }
 
     /* ========== MUTATIVE FUNCTIONS ========== */
 
-    function _requireMinimumBid(uint bid) internal pure returns (uint) {
-        require(bid >= _MINIMUM_BID || bid == 0, "Balance < $0.01");
-        return bid;
+    function _requireMinimumAmount(uint amount) internal pure returns (uint) {
+        require(amount >= _MINIMUM_AMOUNT || amount == 0, "Balance < $0.01");
+        return amount;
     }
 
-    // This must only be invoked during bidding.
-    function bid(address bidder, uint newBid) external onlyMarket {
-        bidOf[bidder] = _requireMinimumBid(bidOf[bidder].add(newBid));
-        totalBids = totalBids.add(newBid);
-    }
-
-    // This must only be invoked during bidding.
-    function refund(address bidder, uint newRefund) external onlyMarket {
-        // The safe subtraction will catch refunds that are too large.
-        bidOf[bidder] = _requireMinimumBid(bidOf[bidder].sub(newRefund));
-        totalBids = totalBids.sub(newRefund);
-    }
-
-    // This must only be invoked after bidding.
-    function claim(
-        address claimant,
-        uint price,
-        uint depositsRemaining
-    ) external onlyMarket returns (uint optionsClaimed) {
-        uint _bid = bidOf[claimant];
-        uint claimable = _claimableBalanceOf(_bid, price, depositsRemaining);
-        // No options to claim? Nothing happens.
-        if (claimable == 0) {
-            return 0;
-        }
-
-        totalBids = totalBids.sub(_bid);
-        bidOf[claimant] = 0;
-
-        totalSupply = totalSupply.add(claimable);
-        balanceOf[claimant] = balanceOf[claimant].add(claimable); // Increment rather than assigning since a transfer may have occurred.
-
-        emit Transfer(address(0), claimant, claimable);
-        emit Issued(claimant, claimable);
+    function mint(address minter, uint amount) external onlyMarket {
+        _requireMinimumAmount(amount);
+        totalSupply = totalSupply.add(amount);
+        balanceOf[minter] = balanceOf[minter].add(amount); // Increment rather than assigning since a transfer may have occurred.
 
-        return claimable;
+        emit Transfer(address(0), minter, amount);
+        emit Issued(minter, amount);
```

Comment:
There is no longer a concept of claiming. For every sUSD that is deposited the user gets 1 long option and 1 short option (minus the 1% minting fee)